File: app/src/main/java/bassamalim/hidaya/database/TelawatVersionsDao.java
package bassamalim.hidaya.database;

import androidx.room.Dao;
import androidx.room.Query;

import java.util.List;

@Dao
public interface TelawatVersionsDao {

    @Query("SELECT * FROM telawat_versions")
    List<TelawatVersionsDB> getAll();

    @Query("SELECT * FROM telawat_versions WHERE reciter_id = :id AND rewaya = :rewaya")
    TelawatVersionsDB getVersion(int id, String rewaya);

    @Query("SELECT suras FROM telawat_versions WHERE reciter_id = :id AND rewaya = :rewaya")
    String getSuras(int id, String rewaya);

}

File: app/src/main/java/bassamalim/hidaya/core/data/database/daos/TelawatVersionsDao.kt
package bassamalim.hidaya.core.data.database.daos

import androidx.room.Dao
import androidx.room.Query
import bassamalim.hidaya.core.data.database.dbs.TelawatVersionsDB

@Dao
interface TelawatVersionsDao {
    @get:Query("SELECT * FROM telawat_versions")
    val all: List<TelawatVersionsDB>

    @Query(
        "SELECT * FROM telawat_versions " +
                "WHERE reciter_id = :reciter_id AND version_id = :version_id"
    )
    fun getVersion(reciter_id: Int, version_id: Int): TelawatVersionsDB

    @Query(
        "SELECT suar FROM telawat_versions " +
                "WHERE reciter_id = :reciter_id AND version_id = :version_id"
    )
    fun getSuar(reciter_id: Int, version_id: Int): String
}